### PR TITLE
Inferred spans for Lambda function URLs

### PR DESCRIPTION
**Pull request by astuyve**
### What does this PR do?

Supports inferred spans for lambda function URLs

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [x] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: src/trace/span-inferrer.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const path = event.rawPath;
+    options.tags = {
+      operation_name: "aws.lambda.url",
+      "http.url": domain + path,
+      endpoint: path,
+      "http.method": event.requestContext.http.method,
+      resource_names: domain + path,
+      request_id: context?.awsRequestId,
+      "span.type": "http",
+      "resource.name": domain + path,
```
Comment (hghotra) [inline src/trace/span-inferrer.ts:30]: 💯

File: src/trace/listener.ts [hunk 2]
```diff
@@ -147,8 +159,12 @@ export class TraceListener {
         ...this.stepFunctionContext,
       };
     }
-
-    if (parentSpanContext !== null) {
+    if (this.inferredSpan !== (null || undefined)) {
```
Comment (hghotra) [inline src/trace/listener.ts:162]: Is there a reason for writing it this way? This basically reduces to:

```
this.inferredSpan !== undefined
```

Perhaps, it should just be:

```suggestion
    if (this.inferredSpan) {
```

**Review by hghotra — APPROVED** [review]
👍🏼

**Comment by astuyve** [comment]
Yup! My bad, this was not supposed to be in the change list

**Pull request merged**